// services-gallery/gallery.js
var rp = require('request-promise');
var sleep = require('sleep');
var idx = 0;
var flood_cnt = 100;
var flood_wait = 16;
function FloodControl()
{
	idx += 1;
	if(idx > 100)
	{
		idx = 0;
		sleep.sleep(flood_wait);
	}
}
function GalleryToken()
{
	var options = {
	    method: 'POST',
	    uri: 'http://gallery.gmancoder.com/oauth/token',
	    form: {
	        "grant_type": "password",
	        "username": "",
	        "password": "",
	        "client_id": "",
	        "client_secret": ""
	    }
	};

	return rp(options)
	    .then(function (body) {
	        //JSON.parse(body);
	        //console.log(JSON.parse(body).access_token);
	        return JSON.parse(body).access_token;
	    })
	    .catch(function (err) {
	        return "";
	    });
}

function GetAlbumImages(album_id)
{
	return GalleryToken().then(function(token) {
		if(token != "") {
			FloodControl();
			var data = {'conditions': [{'field': 'ALBUM_ID', 'operator': "=", "values": [album_id]}]};
			var options = {
			    method: 'POST',
			    uri: 'http://gallery.gmancoder.com/images/get',
			    body: data,
			    headers: {
			    	'Authorization': 'Bearer ' + token
			    },
			    json: true
			};

			return rp(options)
			    .then(function (body) {
			    	console.log(body.Results.images);
			        return body.Results.images;
			    })
			    .catch(function (err) {
			        return "";
			    });
		}
	});
}

function GetThumbnailImage(album_id)
{
	return GetAlbumImages(album_id).then(function(images) {

	});
}

module.exports = {
	GetAlbumImages: GetAlbumImages,
	GetThumbnailImage: GetThumbnailImage
}
