// services-dynamics/dynamics.js
var rp = require('request-promise');
function DynamicsToken()
{
	var options = {
	    method: 'POST',
	    uri: 'http://dynservice.gmancoder.com/v3/token',
	    form: {
	        "grant_type": "password",
	        "username": "gbrewer",
	        "password": "!Pass248word"
	    }
	};

	return rp(options)
	    .then(function (body) {
	        //JSON.parse(body);
	        //console.log(JSON.parse(body).access_token);
	        return JSON.parse(body).access_token;
	    })
	    .catch(function (err) {
	        return "";
	    });
}

function PostToDynamics(token, method, data)
{
	var options = {
	    method: 'POST',
	    uri: 'http://dynservice.gmancoder.com/v3/api/' + method,
	    body: data,
	    headers: {
	    	"content-type": "application/json",
	    	"accept": "application/json",
	    	"authorization": "Bearer " + token
	    },
	    json: true
	};

	return rp(options)
	    .then(function (body) {
	    	return body;
	    })
	    .catch(function (err) {
	    	console.log(err);
	        return "";
	    });
}

module.exports = {
	DynamicsToken: DynamicsToken,
	PostToDynamics: PostToDynamics
}
